src/lib/features/dependent-features/dependent-features-read-model-type.ts

```diff
@@ -1,11 +1,15 @@
 import { IDependency } from '../../types';
+import { FeatureDependency } from './dependent-features';
 
 export interface IDependentFeaturesReadModel {
     getChildren(parents: string[]): Promise<string[]>;
     // given a list of parents and children verifies if some children would be orphaned after deletion
     // we're interested in the list of parents, not orphans
     getOrphanParents(parentsAndChildren: string[]): Promise<string[]>;
     getParents(child: string): Promise<IDependency[]>;
+    getFeatureDependenciesByChildren(
```

Comment:
The naming is inconsistent with other methods. Non of the other methods uses Feature in the method name. I think you can modify getParents to getParent(children: string[]) to make it more generic without introducing a new method